tests/c_languages/CPPCheckBearTest.py

```diff
@@ -30,19 +32,46 @@ def get_results(self, files_to_check):
         self.uut = CPPCheckBear(self.file_dict,
                                 self.section,
                                 self.queue)
-        return list(self.uut.run_bear_from_section([], {}))
 
-    def test_results_complete(self):
+        result = self.uut.run_bear_from_section([], {})
+        if result is not None:
+            return list(result)
+        return
+
+    def test_results_complete_language_cpp(self):
         self.section.append(Setting('enable', 'unusedFunction'))
+        self.section.append(Setting('language', 'c++'))
```

Comment:
Missing `self.section.language = Language[<language_name>]`
Please check other bears tests to get more information on this. As well suggesting you to have a look what does `run_section_from_bear` does! You can find this function definition in [coala](https://github.com/coala/coala) repository.

Reply:
@KVGarg, do we add a `Language` type parameter using `self.section.language = Language[<language_name>]` instead of `self.section.append(Setting('language', 'c++'))` or alongwith `self.section.append(Setting('language', 'c++'))`?

Reply:
along with the line you’ve already added to the new tests!

Reply:
Oh! Thanks